### PR TITLE
Local lockdown postcode input

### PR DESCRIPTION
**Pull request by Rosa-Fox**
[Trello](https://trello.com/c/o5EzGNg9/832-add-a-page-to-ask-users-for-their-postcode)

[Replaces work on this branch](https://github.com/alphagov/finder-frontend/compare/local-lockdown-postcode-input?expand=1)

## What?

- Adds a page to ask users for their postcode.
- This PR just sets up the show view, the following PR will implement posting the inputted data.
- A constraint has been added to the route to return a 404 in production. We will remove this when the feature is ready to be released. 

Co-authored-by: @leenagupte @GDSNewt @JonathanHallam

### CODE REVIEW

File: app/assets/stylesheets/application.scss [hunk 1]
```diff
@@ -52,6 +53,8 @@ $covid-yellow: #fff500;
 @import "views/taxons";
 @import "views/organisations";
 @import "views/organisation";
+@import "views/postcode-lookup";
 @import "views/covid";
 
+
```
Comment (DilwoarH) [inline app/assets/stylesheets/application.scss:59]: I think extra space can be removed.

File: app/assets/stylesheets/application.scss [hunk 2]
```diff
@@ -52,6 +53,8 @@ $covid-yellow: #fff500;
 @import "views/taxons";
 @import "views/organisations";
 @import "views/organisation";
+@import "views/postcode-lookup";
```
Comment (DilwoarH) [inline app/assets/stylesheets/application.scss:56]: Should we move this into `shared/postcode-lookup` to keep it consistent with the template

**Review by leenagupte — COMMENTED** [review]
This is looking good!

There are a few linting issues in the individual commits that I'm choosing to ignore for time 😱 

Otherwise there are a couple of minor things that need to be fixed:
- The title of the page
- The breadcrumbs:

Actual:
Screenshot 2020-10-08 at 15 01 29

Prototype:
Screenshot 2020-10-08 at 15 01 18


It's very close though!

File: app/assets/stylesheets/application.scss (hunk 1 above)
Comment (Rosa-Fox) [reply app/assets/stylesheets/application.scss:59]: ooh yeah! i'll take it out :)

File: app/views/coronavirus_local_restrictions/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+      <% content_for :title, "Collections" %>
+
+      <%= render "govuk_publishing_components/components/title", title: t("coronavirus_local_restrictions.lookup.title") %>
+
+      <%= render "govuk_publishing_components/components/inset_text", text: t('coronavirus_local_restrictions.lookup.inset_text') %>
+
+      <%= sanitize(t("coronavirus_local_restrictions.lookup.body_content")) %>
+
+      <%= render "shared/postcode-lookup" %>
```
Comment (DilwoarH) [inline app/views/coronavirus_local_restrictions/show.html.erb:19]: do we have the error message yet to pass into this?

Comment (leenagupte) [reply app/views/coronavirus_local_restrictions/show.html.erb:19]: I think that will come when we add the POST route.

File: config/locales/en/lookup.yml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+en:
+  coronavirus_local_restrictions:
+    lookup:
+      input_label: Enter your postcode
+      hint_text: For example SW1A 2AA
+      button_text: Find
+      link_text: Find a postcode on Royal Mail's postcode finder
+      href: http://www.royalmail.com/find-a-postcode
+      title: Find out the coronavirus restrictions in a local area
+      inset_text: There are different restrictions in Scotland, Wales and Northern Ireland.
```
Comment (leenagupte) [inline config/locales/en/lookup.yml:10]: As the postcode-lookup view is completely separate from the rest of the coronavirus templates, there's probably and argument for splitting these into their own locale file.

File: app/views/coronavirus_local_restrictions/show.html.erb (hunk 3 above)
Comment (DilwoarH) [reply app/views/coronavirus_local_restrictions/show.html.erb:19]: oh yes - are we adding in the form in another PR?

File: app/assets/stylesheets/application.scss (hunk 2 above)
Comment (Rosa-Fox) [reply app/assets/stylesheets/application.scss]: thanks for moving!

File: app/views/coronavirus_local_restrictions/show.html.erb [hunk 5]
```diff
@@ -7,4 +7,9 @@
 
 <% content_for :title, "Collections" %>
```
[the remark this answers is not in the document]
Comment (Rosa-Fox) [reply app/views/coronavirus_local_restrictions/show.html.erb]: ah yeah 🤦‍♀️ have amended

File: app/views/coronavirus_local_restrictions/show.html.erb (hunk 3 above)
Comment (leenagupte) [reply app/views/coronavirus_local_restrictions/show.html.erb:19]: Yes. This ticket: https://trello.com/c/g3eVg2yg/833-display-the-lockdown-results

Comment (Rosa-Fox) [reply app/views/coronavirus_local_restrictions/show.html.erb:19]: yeah :) next task: https://trello.com/c/g3eVg2yg/833-display-the-lockdown-results

**Comment by DilwoarH** [comment]
> This is looking good!
> 
> There are a few linting issues in the individual commits that I'm choosing to ignore for time 😱
> 
> Otherwise there are a couple of minor things that need to be fixed:
> 
> * The title of the page
> * The breadcrumbs:
> 
> Actual:
> Screenshot 2020-10-08 at 15 01 29
> 
> Prototype:
> Screenshot 2020-10-08 at 15 01 18
> 
> It's very close though!

Should we have the breadcrumb like:

Home > Coronavirus

**Comment by leenagupte** [comment]
> > This is looking good!
> > There are a few linting issues in the individual commits that I'm choosing to ignore for time 😱
> > Otherwise there are a couple of minor things that need to be fixed:
> > 
> > * The title of the page
> > * The breadcrumbs:
> > 
> > Actual:
> > Screenshot 2020-10-08 at 15 01 29
> > Prototype:
> > Screenshot 2020-10-08 at 15 01 18
> > It's very close though!
> 
> Should we have the breadcrumb like:
> 
> Home > Coronavirus

I think it should be:

Home > Coronavirus (COVID-19)

File: config/locales/en/lookup.yml (hunk 4 above)
Comment (Rosa-Fox) [reply config/locales/en/lookup.yml:8]: ah I didn't put in its own file, I put it under a separate key 😅 ... is that ok?

Comment (leenagupte) [reply config/locales/en/lookup.yml:8]: Yeah. Considering the time constraints it's a definite nitpick. We can separate the files when things have calmed down a bit.

File: app/controllers/coronavirus_local_restrictions_controller.rb [hunk 6]
```diff
@@ -1,4 +1,25 @@
...
+  # Breadcrumbs for this page are hardcoded because it doesn't yet have a
+  # content item with parents.
+  def breadcrumbs
+    [
+      {
+        title: "Home",
+        url: "/",
+      },
+      {
+        title: "Coronavirus(COVID-19)",
```
Comment (leenagupte) [inline app/controllers/coronavirus_local_restrictions_controller.rb:24]:
```suggestion
        title: "Coronavirus (COVID-19)",
```
Nitpick!

**Review by leenagupte — COMMENTED** [review]
Two tiny nitpicks left about the title and breadcrumb, otherwise it looks amazing!

File: app/controllers/coronavirus_local_restrictions_controller.rb (hunk 6 above)
Comment (Rosa-Fox) [reply app/controllers/coronavirus_local_restrictions_controller.rb]: ahh! thanks 👍 :)

**Review by leenagupte — APPROVED** [review]
Beautiful! 🎉 💯 👍 🥇 
Screenshot 2020-10-08 at 16 26 18

**Pull request merged**